### PR TITLE
Auth Next in Core build

### PR DESCRIPTION
**Pull request by dmkozh**
# Description

Support for Auth Next in the Core build.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/rust/src/contract.rs [hunk 1]
```diff
@@ -33,7 +34,7 @@ impl From<CxxLedgerInfo> for LedgerInfo {
             protocol_version: c.protocol_version,
             sequence_number: c.sequence_number,
             timestamp: c.timestamp,
-            network_passphrase: c.network_passphrase,
+            network_id: c.network_id.try_into().unwrap(),
```
Comment (sisuresh) [inline src/rust/src/contract.rs:37]: Is panicking here safe? Could we avoid this by storing a `[u8; 32]` in `CxxLedgerInfo`?

File: src/rust/src/lib.rs [hunk 2]
```diff
@@ -68,7 +68,7 @@ mod rust_bridge {
         pub protocol_version: u32,
         pub sequence_number: u32,
         pub timestamp: u64,
-        pub network_passphrase: Vec<u8>,
+        pub network_id: Vec<u8>,
```
Comment (sisuresh) [inline src/rust/src/lib.rs:71]: Should this type be `[u8; 32]`?

File: src/rust/src/contract.rs (hunk 1 above)
Comment (dmkozh) [reply src/rust/src/contract.rs:37]: Yeah, it should be safe (unless there is some bug in core which causes us passing an incorrect value, at which point panicing is fine).

File: src/rust/src/lib.rs (hunk 2 above)
Comment (dmkozh) [reply src/rust/src/lib.rs:71]: Probably, but I don't want to mess with the bridge code to pass arrays.

Comment (sisuresh) [reply src/rust/src/lib.rs:71]: Issue opened related to this - https://github.com/stellar/rs-soroban-env/issues/671.

**Comment by sisuresh** [comment]
r+ 318465b553cb119838276c843a5643fde09524de

**Pull request merged**